crates/cracker/src/lib.rs

```diff
@@ -12,12 +17,57 @@ impl PDFCracker {
     }
 }
 
-impl PDFCracker {
+type ObjectCache = SimpleCache<Result<AnySync, Arc<PdfError>>>;
+type StreamCache = SimpleCache<Result<Arc<[u8]>, Arc<PdfError>>>;
+
+pub struct PDFCrackerState(
+    Storage<Vec<u8>, ObjectCache, StreamCache>
+);
+
+impl PDFCrackerState {
+    /// Init `pdf::file::Storage` so we can reuse it on each `attempt`.
+    pub fn from_cracker(pdf_file: &PDFCracker) -> anyhow::Result<Self> {
+        let res = Storage::with_cache(
+            pdf_file.0.clone(),
+            ParseOptions::strict(),
+            SimpleCache::new(),
+            SimpleCache::new(),
+        );
+
+        match res {
+            Ok(storage) => Ok(Self(storage)),
+            Err(err) => Err(anyhow!(err).context("Failed to init cracker")),
```

Comment:
Not sure if it worth bringing  `anyhow` here or it is better to create a custom error with `thiserror`.